File: ClientApp/src/app/model/Room.ts
import { Deck } from "./Deck";
import { Game, Card, Sentence } from "./Game";
import { Message } from "./Message";
import { User } from "./User";

export class Room {
  id : string = "";
  name : string = "";
  maxPlayers : number = 5;
  state: RoomState = RoomState.Configuration;
  
  ownerId : string = "";
  ownerName : string = "";
  
  sentences : Sentence[] = [];
  cards : Card[] = [];
  users : User[] = [];
  bannedUsers : string[] = [];  
  messages : Message[] = [];
  decks : Deck[] = [];
}

export enum RoomState {
  Configuration,
  Ingame,
  Finished
}

export { Message }

File: Model/Entities/Room.cs
using CardsAngular.Model.Enumarations;

namespace CardsAngular.Model.Entities;

public class Room
{
    public Guid Id { get; set; }
    public string? Name { get; set; }
    public Guid OwnerId { get; set; }
    public string OwnerName { get; set; } = "";
    public int MaxPlayers { get; set; } = 10;
    public List<User> Users { get; set; } = [];
    public List<Guid> BannedUsers { get; set; } = [];
    public List<Deck> Decks { get; set; } = [];
    public List<Sentence> Sentences { get; set; } = [];
    public List<Card> Cards { get; set; } = [];
    public RoomState State { get; set; } = RoomState.New;
    public Game? Game { get; set; }    
}